src/Bicep.Types/Concrete/ResourceType.cs

```diff
@@ -1,21 +1,34 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+using System;
 
 namespace Azure.Bicep.Types.Concrete
 {
+    [Flags]
+    public enum ResourceFlags
+    {
+        None = 0,
+        ReadOnly = 1 << 0,
+        WriteOnly = 1 << 1,
+        Async = 1 << 2,
```

Comment:
I'm a little conflicted on whether we should surface this information to end-users (async vs sync) - to me it feels like a leaky abstraction in the Deployment engine. Is this something we could fix in the Deployment engine, and has there been any discussion on what a fix would look like if so?

To complicate things more, there are RPs which will dynamically decide whether to use an async or sync pattern for a given PUT depending on the complexity of the work required, so it may be difficult to be fully accurate with trying to identify in Bicep.

Reply:
The rationale for adding this tag was https://github.com/Azure/bicep-types-az/pull/780#discussion_r864221615, though I think you may be right about this tag capturing and surfacing a leaky abstraction. As I understand the ARM deployment process, the deployment engine will discard any `PUT` response for long-running operations, so there would be no distinction between the behavior exhibited by various `reference()` invocation syntaxes.

This tag was added because it *may* enable a Bicep compiler optimization, but it merits more discussion and is not needed in this PR.